schemas/preconversion_assessment_schema_1.0.json

```diff
@@ -0,0 +1,41 @@
+{
+    "title": "Convert2rhel pre-conversion script schema",
+    "description": "Script is expected to set up Convert2rhel and run pre-conversion assessment. This schema defines the output format that is expected by Red Hat Insights Task UI.",
+    "type": "object",
+    "properties": {
+      "status": {
+        "$ref": "#/$defs/status_codes"
+      },
+      "report": {
+        "type": "string"
+      },
+      "message": {
+        "type": "string"
+      },
+      "report_json": {
+        "type": "object",
+        "properties": {
+          "entries": {
+            "type": "object"
+          }
+        },
+        "required": ["entries"]
+      }
+    },
+    "required": ["status", "message", "report", "report_json"],
+    "additionalProperties": false,
+    "$defs": {
+      "status_codes": {
+          "description": "The severity of the results and messages",
+          "type": "string",
+          "enum": [
+              "SUCCESS",
+              "INFO",
+              "WARNING",
+              "SKIP",
+              "OVERRIDABLE",
+              "ERROR"
```

Comment:
@r0x0d I took theese from the [convert2rhel schema](https://github.com/oamg/convert2rhel/pull/879/files#diff-450cf3ddd7edd0b5bbc1c454d3db12f6fb1e7648bdb82a09236995642982a6cbR40), is the INFO status missing on purpose in the script code or is it an oversight?

Reply:
@abadger had an idea sometime ago that we could import convert2rhel status in the script by literally doing and `from convert2rhel.actions import STATUS_CODE`, and that would save us from missing any status code we would eventually miss... But that would not solve the problem with the schema. 

For now, I would say that the status that we have mapped here are enough for now, and if there is any new statuses introduced in convert2rhel, we can make this update a dependency of the task and update this script as well.

And also, I would say that it will be sometime for convert2rhel to add a new status that is not mapped anywhere...

Reply:
Also, I'm not quite skilled with json schema, but I think you can import schemas from other URLs, like the one from convert2rhel and use the properties from that schema? This way we would say that whatever is the default schema for convert2rhel regarding statuses, we will be using the same.

Not sure if that's a thing, but if there is something like that, maybe it would be worth it to use.

Reply:
I mean that could work, but it would be again pain to maintain the correct link, if the status would change tchema on convert2 rhel side as well and then I would have to update the url anyway .. I would just keep it as it is now, as you said it is just dependency that has to be taken care of if the changes are made 

And it can live in jira in some minor task if needed, to try to automate it somehow :shrug:

Reply:
Sounds good to me.